### ISSUE TITLE
Indexing Severely Slows VSCode for repos with large numbers of non-local Modules

### ISSUE THREAD

**Issue by christek91** [mauve/vscode-terraform#153]
A bit of context: my team has a repository (lets call it Library), which contains a large number of our Terraform Modules. 

We actually use these modules in a number of different Terraform Configs that live in a separate repository (lets call it Infrastructure). There are multiple Terraform projects in this repo, each of which uses a number of the Modules that live in the Library repo.

The way that the TF code in Infrastructure uses these modules is by using
```
module "my-module-instance"{
  source = "git::https://gitrepo.com/Library.git//path/to/my-module"
  ...
}
```

On `terraform init`, terraform clones from that `source` repo for each of the modules declared in the config. The end result of this is that we have dozens of copies of the Library repository in various `.terraform` directories in the `Infrastructure` repo, each of which contains dozens of of Terraform files, for a grand total of (dozens x dozens) of Terraform files for this plugin to index. 

So many that it regularly freezes VSCode on my Macbook Pro, and consumes an entire CPU core/2GB of memory. It's been very good for keeping my apartment warm this winter ;)

So that's the issue. I think there is a relatively easy fix for this though.

Generally speaking, none of the files in `.terraform` should refer to any of the files outside of it, since they're all dependencies of the Configuration that has cloned them. It's also pretty unlikely that most users would want to edit the files that are stored in `.terraform`. 

Additionally, I don't think that the current codelens/indexing implementation exposes any module-interdependencies. The reference counts and links that codelens shows are all localized within the current module as far as I can tell. 

Therefore, most users probably don't need to index the files in `.terraform` directories at all, since the index of those files would only be useful to users editing those files, and that's not a common practice. This would save a lot of CPU time spent indexing files for no real use.

Perhaps this plugin could expose a config value that lets users ignore a list of directories during indexing ala `.gitignore`, and maybe even make `.terraform` a default value.

**Comment by LoboHacks** [comment]
I have the same issue.
I even have to kill the terraform process several times when i launch my workspace to get out of the infinite indexing loop of hell...

**Comment by MattMencel** [comment]
The default setting for `terraform.indexing` is this...

```
    "terraform.indexing": {
        "enabled": true,
        "liveIndexing": true,
        "delay": 500,
        "exclude": [
            ".terraform/**/*",
            "**/.terraform/**/*"
        ]
    },
```

That second exclude should force it to not pay attention to any `.terraform` directory anywhere, but I'm finding the same behavior.  If I open a folder in VSCode with lots of Terraform directories, this plugin chokes trying to index the contents of all the `.terraform` folders.

**Comment by christek91** [comment]
The issue that I was experienced should be fixed with the next release (1.3.8). I see that @mauve made a release commit, so I'd expect it should show up in the vscode extension marketplace in the near future!

**Comment by mauve** [comment]
@christek91 thanks for your help, just published the release.

Thanks for using my plugin, feel free to reopen if 1.3.8 does not fix this issue.
